rust/src/delta.rs

```diff
@@ -540,8 +550,35 @@ impl DeltaTable {
             self.version(),
         );
 
-        while let PeekCommit::New(new_version, actions) =
-            self.peek_next_commit(self.version()).await?
+        // construct stream yielding (version, bytes)
+        let store = self.storage.clone();
+        let log_stream = futures::stream::iter(self.version() + 1..).map(|i| {
+            let store2 = store.clone();
+            let loc = commit_uri_from_version(i).clone();
+            async move {
+                let out = store2.get(&loc);
+                (i, out.await)
+            }
+        });
+
+        let buf_size = 50; // TODO
+
+        // why is mut needed here and is it OK?
+        let mut log_buffer = {
+            match max_version {
+                Some(n) => log_stream.take((n - self.version() + 2).try_into().unwrap()).buffered(buf_size),
```

Comment:
```suggestion
                Some(n) => log_stream.take((n - self.version() + 2).try_into()?).buffered(buf_size),
```

I'm not sure off the top of my head, but I think the `?` will work and help ensure that errors get propagated up to the `DeltaResult` returned by the function

Reply:
This didn't work directly, but I added explicit handling for the error cases